pkg/utils/utils.go

```diff
@@ -60,6 +61,20 @@ func GetPfAddr(pciAddr string) (string, error) {
 // GetPfName returns SRIOV PF name for the given VF
 // If device is not VF then it will return its own ifname if exist else empty string
 func GetPfName(pciAddr string) (string, error) {
+	pfEswitchMode, err := GetPfEswitchMode(pciAddr)
```

Comment:
@Eoghan1232 @zshi-redhat for intel nics, i assume we can query the eswitch mode via devlink. can you approve that is the case ?

else we would need to log a warning and fallback on error

Reply:
I tried on the i40e device, got the following error:
```
W1110 00:36:16.715844 2172814 pciNetDevice.go:81] unable to get PF name "error getting devlink device attributes for net device 0000:3b:00.0 no such device"
```

I suppose the eswitch mode query is not supported.
@Eoghan1232  Could you help to confirm?

Reply:
getting an error with `devlink` tool as well ?

```
root ~ $ devlink dev show
pci/0000:03:00.0
pci/0000:03:00.1
root ~ $ devlink dev eswitch show pci/0000:03:00.0
pci/0000:03:00.0: mode switchdev inline-mode none encap enable
```

Reply:
Yes, error with devlink :
```
# devlink dev eswitch show pci/0000:3b:00.0
devlink answers: No such device
```

Reply:
ack, thanks for checking this ! I think some intel NIC do support devlink as we got PR #387 
So, i suggest we check for error (ENODEV) and fallback to previous logic.

Reply:
Will test this today and reply back.

Reply:
Hi @adrianchiris @zshi-redhat 

@ipatrykx has tested this in his env.
It seems currently it is only supported by 800 series Nics and not 700 series Nics.

```
root ~ $ devlink dev eswitch show pci/0000:83:00.0
pci/0000:83:00.0: mode switchdev
root ~ $ devlink dev eswitch show pci/0000:81:00.0
devlink answers: No such device
```
the first card is an 800 series 
the second one is a 700 series with i40e

We also updated i40e to latest 2.17.4 driver and downgraded 800 series to 1.5.8 and it's still the same

```
root ~ $ devlink dev eswitch show pci/0000:81:00.0
devlink answers: No such device
root ~ $ devlink dev eswitch show pci/0000:83:00.0
pci/0000:83:00.0: mode legacy
```

Reply:
Thanks for checking this out!

we should do fallback if device does not support devlink ("no such device" error). this would be good for other vendor's that dont support devlink.

Reply:
Thanks for testing the changes! Added a check for error message and fallback to the default impl if device doesn't support eswitch mode query